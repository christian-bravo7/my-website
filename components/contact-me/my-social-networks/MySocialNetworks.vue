<template>
  <section class="flex justify-center items-center">
    <div class="grid gap-6 grid-cols-4 md:grid-cols-2 lg:grid-cols-4 page-social-icons">
      <SocialNetworkIcon
        v-for="({ link, component }, index) in socialMediaIcons"
        :key="index"
        :link="link"
      >
        <Component :is="component" />
      </SocialNetworkIcon>
    </div>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';
import CodepenIcon from '~/assets/img/codepen.svg?inline';
import GithubIcon from '~/assets/img/github.svg?inline';
import InstagramIcon from '~/assets/img/instagram.svg?inline';
import LinkedinIcon from '~/assets/img/linkedin.svg?inline';
import { SocialMediaIcon } from '@/types/social-media.types';

@Component({
  components: {
    CodepenIcon,
    GithubIcon,
    InstagramIcon,
    LinkedinIcon,
  },
})
export default class MySocialNetworks extends Vue {
  socialMediaIcons: Array<SocialMediaIcon> = [
    { link: 'https://codepen.io/christian-bravo', component: CodepenIcon },
    { link: 'https://github.com/christian-bravo7', component: GithubIcon },
    { link: 'https://www.instagram.com/christian.bravo_/', component: InstagramIcon },
    { link: 'www.linkedin.com/in/bravo-christian', component: LinkedinIcon },
  ]
}
</script>
