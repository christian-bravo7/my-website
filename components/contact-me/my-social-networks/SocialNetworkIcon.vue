<template>
  <a
    target="_blank"
    :href="link"
    class="w-8 h-8 md:w-12 md:h-12 | p-1 | bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-500 hover:text-gray-50 hover:border-pink-500 rounded-full social-icon"
  >
    <slot />
  </a>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';

@Component
export default class SocialNetworkIcon extends Vue {
  @Prop({ type: String, required: true })
  readonly link!: string;
}
</script>

<style lang="scss" scoped>
.social-icon {
  transition-duration: 300ms;
  transition-property: background-color, box-shadow;

  &:hover {
    box-shadow: 0 0 0 6px theme('colors.pink.100');
  }

  svg {
    transition-duration: 300ms;
    transition-property: fill;

    @apply fill-current;
  }
}

@screen md {
  .social-icon {
    padding: 10px;
  }
}

@include dark-mode {
  .social-icon {
    &:hover {
      box-shadow: 0 0 0 6px theme('colors.blue.600');
    }
  }
}
</style>
