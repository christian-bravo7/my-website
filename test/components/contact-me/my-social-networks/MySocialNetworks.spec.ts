import { mount } from '@vue/test-utils';

import SocialNetworkIcon from '@/components/contact-me/my-social-networks/SocialNetworkIcon.vue';
import MySocialNetworks from '@/components/contact-me/my-social-networks/MySocialNetworks.vue';

describe('MySocialNetworks component', () => {
  it('should render correclty', () => {
    const wrapper = mount(MySocialNetworks);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display 4 social icons', () => {
    const wrapper = mount(MySocialNetworks);

    const SocialNetworkIcons = wrapper.findAllComponents(SocialNetworkIcon);

    expect(SocialNetworkIcons.length).toBe(4);
  });
});
