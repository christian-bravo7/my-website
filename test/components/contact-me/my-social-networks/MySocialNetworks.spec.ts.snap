// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MySocialNetworks component should render correclty 1`] = `
<section class="flex justify-center items-center">
  <div class="grid gap-6 grid-cols-4 md:grid-cols-2 lg:grid-cols-4 page-social-icons"><a target="_blank" href="https://codepen.io/christian-bravo" class="w-8 h-8 md:w-12 md:h-12 | p-1 | bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-500 hover:text-gray-50 hover:border-pink-500 rounded-full social-icon"><svg></svg></a><a target="_blank" href="https://github.com/christian-bravo7" class="w-8 h-8 md:w-12 md:h-12 | p-1 | bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-500 hover:text-gray-50 hover:border-pink-500 rounded-full social-icon"><svg></svg></a><a target="_blank" href="https://www.instagram.com/christian.bravo_" class="w-8 h-8 md:w-12 md:h-12 | p-1 | bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-500 hover:text-gray-50 hover:border-pink-500 rounded-full social-icon"><svg></svg></a><a target="_blank" href="https://www.linkedin.com/in/bravo-christian" class="w-8 h-8 md:w-12 md:h-12 | p-1 | bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-500 hover:text-gray-50 hover:border-pink-500 rounded-full social-icon"><svg></svg></a></div>
</section>
`;
